src/main/java/digital/slovensko/autogram/ui/gui/GUI.java

```diff
@@ -240,14 +298,46 @@ public void onDocumentSaved(File targetFile) {
         stage.show();
     }
 
+    @Override
+    public void onDocumentBatchSaved(BatchGuiResult result) {
+        var controller = new BatchSigningSuccessDialogController(result, hostServices);
+        var root = GUIUtils.loadFXML(controller, "batch-signing-success-dialog.fxml");
+
+        var stage = new Stage();
+        if (result.hasErrors()) {
+            stage.setTitle("Dokumenty boli podpísané");
+        } else {
+            stage.setTitle("Dokumenty boli úspešne podpísané");
+        }
+        stage.setScene(new Scene(root));
+        stage.setResizable(false);
+        stage.sizeToScene();
+        GUIUtils.suppressDefaultFocus(stage, controller);
+        stage.show();
+    }
+
     @Override
     public void onWorkThreadDo(Runnable callback) {
-        new Thread(callback).start();
+        if (DEBUG && !Platform.isFxApplicationThread())
+            throw new RuntimeException("Can be run only on UI thread");
+
+        if (Platform.isFxApplicationThread()) {
+            new Thread(callback).start();
+        } else {
+            callback.run();
+        }
     }
 
     @Override
     public void onUIThreadDo(Runnable callback) {
-        Platform.runLater(callback);
+        if (DEBUG && Platform.isFxApplicationThread())
```

Comment:
Tu sa vyskytujú nejaké problémy?

Reply:
ooo ano,  toto som riesil strasne dlho, niektore veci robili dost sialene veci, bol problem s tym ze "callback-y" sa nespustali spravne a tak podobne. Neviem uplne povedat preco to teraz funguje a predtym nie, ale riesi to aj to ze vznikali chainy threadov work->work->work->ui->ui->work....